lib/core/macros.nim

```diff
@@ -1697,3 +1697,18 @@ proc extractDocCommentsAndRunnables*(n: NimNode): NimNode =
         result.add ni
       else: break
     else: break
+
+proc freshIdentNodes*(n: NimNode): NimNode =
+  ## Replaces every `nnkSym` node in `n` by a fresh identifier node.
+  ## This forces the compiler to perform a new lookup pass.
+  case n.kind:
+  of nnkSym:
+    result = ident(n.repr)
```

Comment:
Are there some reasons using `repr`?

Reply:
oh yes, it should be `$n`

Reply:
As disruptek said on irc: "a symbol created with genSym needs to be rendered as `ident(repr sym)` in order to capture the unique name correctly."

So in the test I added `ident($n)` or similar will output just "tmp" and redefinition error, whereas with `repr` you get `tmp14141` and works properly.

Reply:
AFAIK, with repr you can also get ```tmp`12312``` which will not compile

Reply:
> AFAIK, with repr you can also get `` tmp`12312 `` for gensym which will not compile

test passes so it might be fixed

Reply:
Yep, ```tmp`12312``` is fine to use as an identifier once past the parser

Reply:
Maybe there's a more correct method than `repr`, but I am personally fine with its use here.